cpp/include/Ice/Proxy.h

```diff
@@ -687,282 +926,132 @@ class ICE_API ObjectPrx : public ::std::enable_shared_from_this<ObjectPrx>
      * @param facet The facet for the new proxy.
      * @return A proxy with the new facet.
      */
-    ::std::shared_ptr<::Ice::ObjectPrx> ice_facet(const ::std::string& facet) const;
+    std::shared_ptr<::Ice::ObjectPrx> ice_facet(const ::std::string& facet) const;
 
     /**
      * Obtains the adapter ID for this proxy.
      * @return The adapter ID. If the proxy does not have an adapter ID, the return value is the empty string.
      */
     ::std::string ice_getAdapterId() const;
 
-    /**
-     * Obtains a proxy that is identical to this proxy, except for the adapter ID.
-     * @param id The adapter ID for the new proxy.
-     * @return A proxy with the new adapter ID.
-     */
-    ::std::shared_ptr<::Ice::ObjectPrx> ice_adapterId(const ::std::string& id) const;
```

Comment:
ObjectPrx now use the exact same factory methods as the other proxies; they are provided by Proxy.